src/components/attendee/PortalModal/PortalModal.tsx

```diff
@@ -1,35 +1,101 @@
-import React, { MouseEventHandler } from "react";
-import { Popover, PopoverProps } from "components/attendee/Popover";
+import React, { MouseEventHandler, RefObject } from "react";
+import { useCss } from "react-use";
+import classNames from "classnames";
+import { Button } from "components/attendee/Button";
 
-import { Room } from "types/rooms";
+import {
+  COVERT_ROOM_TYPES,
+  SCSS_SPACE_EMPTY,
+  SCSS_SPACE_PORTAL_EVENT_WIDTH,
+  STRING_DASH_SPACE,
+  STRING_SPACE,
+} from "settings";
 
-import { useSpaceById } from "hooks/spaces/useSpaceById";
+import { RoomType, RoomWithBounds } from "types/rooms";
+import { RoomVisibility } from "types/RoomVisibility";
+import { WorldEvent } from "types/venues";
+
+import { eventEndTime, eventStartTime } from "utils/event";
+import { formatTimeLocalised } from "utils/time";
 
-import { Button } from "../Button";
+import { useSpaceById } from "hooks/spaces/useSpaceById";
 
 import CN from "./PortalModal.module.scss";
 
-interface PortalModalProps extends PopoverProps {
+interface PortalModalProps {
   onEnter?: MouseEventHandler<HTMLButtonElement>;
-  portal: Room;
+  portal: RoomWithBounds;
+  event?: WorldEvent;
+  portalRef: RefObject<HTMLDivElement> | null;
 }
 
+const topOffsetMultiplier = 5.5;
+const leftOffsetDivider = 2;
+
 export const PortalModal: React.FC<PortalModalProps> = ({
-  referenceElement,
   onEnter,
   portal,
+  event,
+  portalRef,
 }) => {
   const { space } = useSpaceById({ spaceId: portal.spaceId });
+  const top =
+    portal.bounds.top + (portal.bounds.height * portal.y_percent) / 100;
+  const height = (portal.bounds.height * portal.height_percent) / 100;
+
+  const { width = 0, left = 0 } =
+    portalRef?.current?.getBoundingClientRect() ?? {};
+
+  const popoverStyle = useCss({
+    top: `${top + (height + SCSS_SPACE_EMPTY * topOffsetMultiplier)}px`,
```

Comment:
```suggestion
    top: `${top + height + topOffset}px`,
```

looks like `SCSS_SPACE_EMPTY * topOffsetMultiplier` is a const.
why 5.5 by the way? wouldn't it be easier if we just use `topOffset = 66px` ? anyway, 5.5 doesn't seem to be clear to me.